### ISSUE TITLE
Setting channel_alias breaks conda

### ISSUE THREAD

**Issue by koverholt** [conda/conda#3054]
Running conda 4.1.8.

If I set `channel_alias` in my `.condarc` file, like:

```
channel_alias: https://conda.anaconda.org/
```

I get errors when I try to install, search, etc.:

```
$ conda search conda
Traceback (most recent call last):
  File "/Users/koverholt/anaconda3/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/Users/koverholt/anaconda3/lib/python3.5/site-packages/conda/cli/main.py", line 61, in main
    from conda.cli import conda_argparse
  File "/Users/koverholt/anaconda3/lib/python3.5/site-packages/conda/cli/conda_argparse.py", line 15, in <module>
    from .common import add_parser_help
  File "/Users/koverholt/anaconda3/lib/python3.5/site-packages/conda/cli/common.py", line 12, in <module>
    from ..config import (envs_dirs, default_prefix, platform, update_dependencies,
  File "/Users/koverholt/anaconda3/lib/python3.5/site-packages/conda/config.py", line 472, in <module>
    load_condarc(rc_path)
  File "/Users/koverholt/anaconda3/lib/python3.5/site-packages/conda/config.py", line 430, in load_condarc
    channel_alias = remove_binstar_tokens(channel_alias.rstrip('/') + '/')
  File "/Users/koverholt/anaconda3/lib/python3.5/site-packages/conda/config.py", line 276, in remove_binstar_tokens
    return BINSTAR_TOKEN_PAT.sub(r'\1', url)
NameError: name 'BINSTAR_TOKEN_PAT' is not defined
```

**Comment by danielfrg** [comment]
I felt all the emojis when I saw this particular issue.

**Comment by kalefranz** [comment]
Bad news, I don't want to fix this until 4.2.  Good news, 4.2 canary drops tomorrow.

**Comment by bheklilr** [comment]
Also had this issue with conda-4.1.8-py35_0 on win-64 installed from the main anaconda channel.  Hope that fix hits the repos tomorrow!

**Comment by koverholt** [comment]
Any updates here? This is blocking an installation for an enterprise customer.

**Comment by bheklilr** [comment]
@koverholt as far as I can tell (without having read the code) this bug only appears when the channel alias is set in the config. By commenting out that line and then providing full urls for all channels this problem can at least be circumvented. I've had to do this since I also use an internal server. It works, but isn't ideal.

**Comment by koverholt** [comment]
Thanks for the info @bheklilr. This is for our enterprise Anaconda platform installer, we need to be able to use `channel_alias` as part of our installation process so that we don't have to pass full channel URLs around everywhere during installation or when users are conda installing packages across a cluster.

**Comment by quasiben** [comment]
This is a fairly severe issue for enterprise customers as @koverholt noted.  To demonstrate the issue, if we install from a channel like so:

```
[root@ip-172-31-54-159 salt]# /opt/continuum/bin/conda --debug install -c anaconda python=2 pyopenssl
```

Conda slowly slowly moves through the following retries when the customer is airgapped.

```
DEBUG:conda.config:Could not import binstar
DEBUG:conda.fetch:channel_urls=OrderedDict([('https://conda.anaconda.org/anaconda/linux-64/', ('anaconda', 1)), ('https://conda.anaconda.org/anaconda/noarch/', ('anaconda', 1)), ('http://ip-172-31-57-193.ec2.internal:8080/conda/anaconda/linux-64/', ('http://ip-172-31-57-193.ec2.internal:8080/conda/anaconda', 2)), ('http://ip-172-31-57-193.ec2.internal:8080/conda/anaconda/noarch/', ('http://ip-172-31-57-193.ec2.internal:8080/conda/anaconda', 2))
Fetching package metadata ...INFO:stdoutlog:Fetching package metadata ...
DEBUG:requests.packages.urllib3.util.retry:Converted retries value: 3 -> Retry(total=3, connect=None, read=None, redirect=None)
INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): conda.anaconda.org
DEBUG:requests.packages.urllib3.util.retry:Incremented Retry for (url='/anaconda/linux-64/repodata.json'): Retry(total=2, connect=None, read=None, redirect=None)
WARNING:requests.packages.urllib3.connectionpool:Retrying (Retry(total=2, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f8155761a50>: Failed to establish a new connection: [Errno 110] Connection timed out',)': /anaconda/linux-64/repodata.json
INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (2): conda.anaconda.org
DEBUG:requests.packages.urllib3.util.retry:Incremented Retry for (url='/anaconda/linux-64/repodata.json'): Retry(total=1, connect=None, read=None, redirect=None)
WARNING:requests.packages.urllib3.connectionpool:Retrying (Retry(total=1, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f8155761bd0>: Failed to establish a new connection: [Errno 110] Connection timed out',)': /anaconda/linux-64/repodata.json
INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (3): conda.anaconda.org
DEBUG:requests.packages.urllib3.util.retry:Incremented Retry for (url='/anaconda/linux-64/repodata.json'): Retry(total=0, connect=None, read=None, redirect=None)
WARNING:requests.packages.urllib3.connectionpool:Retrying (Retry(total=0, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f8155761d50>: Failed to establish a new connection: [Errno 110] Connection timed out',)': /anaconda/linux-64/repodata.json
INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (4): conda.anaconda.org
Could not connect to https://conda.anaconda.org/anaconda/linux-64/
INFO:stderrlog:Could not connect to https://conda.anaconda.org/anaconda/linux-64/

DEBUG:conda.fetch:Connection error: HTTPSConnectionPool(host='conda.anaconda.org', port=443): Max retries exceeded with url: /anaconda/linux-64/repodata.json (Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f8155761e90>: Failed to establish a new connection: [Errno 110] Connection timed out',)): https://conda.anaconda.org/anaconda/linux-64/

.DEBUG:dotupdate:fetching repodata: args ('https://conda.anaconda.org/anaconda/linux-64/',) kwargs {'session': <conda.connection.CondaSession object at 0x7f81557c8f10>, 'use_cache': False}
INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (5): conda.anaconda.org
```

**Comment by koverholt** [comment]
This is fixed with conda 4.1.9, thank you!

**Comment by HugoTian** [comment]
Yes conda 4.1.9 release today, and it fix this issue.

For more detail, please refer #3104

I will close this issue, and if there is still problem, feel free to reopen it and @ me
